flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThread.java

```diff
@@ -22,12 +22,16 @@
 import org.apache.flink.annotation.VisibleForTesting;
 import org.apache.flink.api.java.tuple.Tuple2;
 import org.apache.flink.metrics.MetricGroup;
+import org.apache.flink.streaming.connectors.kafka.config.RateLimiterFactory;
 import org.apache.flink.streaming.connectors.kafka.internals.ClosableBlockingQueue;
 import org.apache.flink.streaming.connectors.kafka.internals.KafkaCommitCallback;
 import org.apache.flink.streaming.connectors.kafka.internals.KafkaTopicPartitionState;
 import org.apache.flink.streaming.connectors.kafka.internals.KafkaTopicPartitionStateSentinel;
 import org.apache.flink.streaming.connectors.kafka.internals.metrics.KafkaMetricWrapper;
 
+import org.apache.flink.shaded.guava18.com.google.common.util.concurrent.RateLimiter;
```

Comment:
Do we want to directly import class from the shaded jar?

Reply:
This import should not be here at all after we find the appropriate abstraction. But in general, we should use the shaded dependency or not?

Reply:
yes, if you want to use something from guava then you should use the shaded dependency. The only exception is if a another dependency exposes a guava-related API.

Reply:
@zentol @tweise Just for my own curiosity. I agree that we should shade the jar to avoid interference with other user imports. What I am wondering is how should that be done. Whether it should be done in the pom.xml or from the import explicitly.  I am not sure about the difference between those two ways. But I did not see such explicit imports in Flink anywhere else. Am I missing something?

Reply:
@becketqin I'm not familiar with shaded dependency but let's say that it is a common case we use such explicit imports. You can just search "import org.apache.flink.shaded.guava18` and there are over 100 use points.